cmd/kubeadm/app/cmd/token.go

```diff
@@ -268,10 +283,72 @@ func RunGenerateToken(out io.Writer) error {
 	return nil
 }
 
+func formatBootstrapToken(obj *outputapiv1alpha1.BootstrapToken) string {
+	ttl := "<forever>"
+	expires := "<never>"
+	if obj.Expires != nil {
+		ttl = duration.ShortHumanDuration(obj.Expires.Sub(time.Now()))
+		expires = obj.Expires.Format(time.RFC3339)
+	}
+	ttl = fmt.Sprintf("%-9s", ttl)
+
+	usages := strings.Join(obj.Usages, ",")
+	if len(usages) == 0 {
+		usages = "<none>"
+	}
+	usages = fmt.Sprintf("%-22s", usages)
+
+	description := obj.Description
+	if len(description) == 0 {
+		description = "<none>"
+	}
+	description = fmt.Sprintf("%-56s", description)
+
+	groups := strings.Join(obj.Groups, ",")
+	if len(groups) == 0 {
+		groups = "<none>"
+	}
+
+	return fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%s\n", obj.Token, ttl, expires, usages, description, groups)
+}
+
+// upgradePlanPrinter prints upgrade plan in a text form
+type tokenTextPrinter struct {
+	output.ResourcePrinterWrapper
+	columns         []string
+	headerIsPrinted bool
+}
+
+// PrintObj is an implementation of ResourcePrinter.PrintObj for plain text output
+func (ttp *tokenTextPrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
+	tabw := tabwriter.NewWriter(writer, 10, 4, 3, ' ', 0)
+
+	// Print header
+	if !ttp.headerIsPrinted {
+		fmt.Fprintln(tabw, strings.Join(ttp.columns, "\t"))
+		ttp.headerIsPrinted = true
+	}
+
+	// Print token
+	fmt.Fprint(tabw, formatBootstrapToken(obj.(*outputapiv1alpha1.BootstrapToken)))
+
+	return tabw.Flush()
+}
+
+type tokenTextPrintFlags struct{}
+
+// ToPrinter returns resource printer for the text output format
+func (tpf tokenTextPrintFlags) ToPrinter(outputFormat string) (output.KubeadmPrinter, error) {
```

Comment:
unless there is demand to copy `tpf` in memory on each call to `ToPrinter`, this should be:
`func (tpf *tokenTextPrintFlags)`.